[Bug] torchcluster installation failed

# 🐛 Bug

torchcluster installation failed @pz-white 

## To reproduce

See https://github.com/pykale/pykale/runs/5524944626?check_suite_focus=true

** Stack trace/error message **
```
Looking in links: https://pytorch-geometric.com/whl/torch-1.7.0+cpu.html
Collecting torch-cluster
  Downloading torch_cluster-1.6.0.tar.gz (43 kB)
Building wheels for collected packages: torch-cluster
  Building wheel for torch-cluster (setup.py): started
  Building wheel for torch-cluster (setup.py): finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: 'C:\Miniconda\envs\test\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pip-install-p2vpppra\\torch-cluster_2f5e4beb6db144f8be0cd3fd139cf[650](https://github.com/pykale/pykale/runs/5524944626?check_suite_focus=true#step:4:650)\\setup.py'"'"'; __file__='"'"'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pip-install-p2vpppra\\torch-cluster_2f5e4beb6db144f8be0cd3fd139cf650\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\runneradmin\AppData\Local\Temp\pip-wheel-061tex3q'
  Running setup.py clean for torch-cluster
       cwd: C:\Users\runneradmin\AppData\Local\Temp\pip-install-p2vpppra\torch-cluster_2f5e4beb6db144f8be0cd3fd139cf650\
  Complete output (74 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win-amd64-3.8
  creating build\lib.win-amd64-3.8\torch_cluster
  copying torch_cluster\fps.py -> build\lib.win-amd64-3.8\torch_cluster
  copying torch_cluster\graclus.py -> build\lib.win-amd64-3.8\torch_cluster
  copying torch_cluster\grid.py -> build\lib.win-amd64-3.8\torch_cluster
  copying torch_cluster\knn.py -> build\lib.win-amd64-3.8\torch_cluster
  copying torch_cluster\nearest.py -> build\lib.win-amd64-3.8\torch_cluster
  copying torch_cluster\radius.py -> build\lib.win-amd64-3.8\torch_cluster
  copying torch_cluster\rw.py -> build\lib.win-amd64-3.8\torch_cluster
  copying torch_cluster\sampler.py -> build\lib.win-amd64-3.8\torch_cluster
  copying torch_cluster\__init__.py -> build\lib.win-amd64-3.8\torch_cluster
  running egg_info
  writing torch_cluster.egg-info\PKG-INFO
  writing dependency_links to torch_cluster.egg-info\dependency_links.txt
  writing requirements to torch_cluster.egg-info\requires.txt
  writing top-level names to torch_cluster.egg-info\top_level.txt
  reading manifest file 'torch_cluster.egg-info\SOURCES.txt'
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-install-p2vpppra\torch-cluster_2f5e4beb6db144f8be0cd3fd139cf650\setup.py", line 98, in <module>
      setup(
    File "C:\Miniconda\envs\test\lib\site-packages\setuptools\__init__.py", line 153, in setup
      return distutils.core.setup(**attrs)
    File "C:\Miniconda\envs\test\lib\distutils\core.py", line 148, in setup
      dist.run_commands()
    File "C:\Miniconda\envs\test\lib\distutils\dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "C:\Miniconda\envs\test\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "C:\Miniconda\envs\test\lib\site-packages\wheel\bdist_wheel.py", line 299, in run
      self.run_command('build')
    File "C:\Miniconda\envs\test\lib\distutils\cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "C:\Miniconda\envs\test\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "C:\Miniconda\envs\test\lib\distutils\command\build.py", line 135, in run
      self.run_command(cmd_name)
    File "C:\Miniconda\envs\test\lib\distutils\cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "C:\Miniconda\envs\test\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\build_py.py", line 46, in run
      self.build_package_data()
    File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\build_py.py", line 103, in build_package_data
      for package, src_dir, build_dir, filenames in self.data_files:
    File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\build_py.py", line 55, in __getattr__
      self.data_files = self._get_data_files()
    File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\build_py.py", line 67, in _get_data_files
      self.analyze_manifest()
    File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\build_py.py", line 121, in analyze_manifest
      self.run_command('egg_info')
    File "C:\Miniconda\envs\test\lib\distutils\cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "C:\Miniconda\envs\test\lib\distutils\dist.py", line 985, in run_command
      cmd_obj.run()
    File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\egg_info.py", line 299, in run
      self.find_sources()
    File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\egg_info.py", line 306, in find_sources
      mm.run()
    File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\egg_info.py", line 541, in run
      self.add_defaults()
    File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\egg_info.py", line 585, in add_defaults
      self.read_manifest()
    File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\sdist.py", line 188, in read_manifest
      self.filelist.append(line)
    File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\egg_info.py", line 483, in append
      path = convert_path(item)
    File "C:\Miniconda\envs\test\lib\distutils\util.py", line 122, in convert_path
      raise ValueError("path '%s' cannot be absolute" % pathname)
  ValueError: path '/Users/rusty1s/github/pytorch_cluster/csrc/fps.cpp' cannot be absolute
  ----------------------------------------
  ERROR: Failed building wheel for torch-cluster
Failed to build torch-cluster
Installing collected packages: torch-cluster
    Running setup.py install for torch-cluster: started
    Running setup.py install for torch-cluster: finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: 'C:\Miniconda\envs\test\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pip-install-p2vpppra\\torch-cluster_2f5e4beb6db144f8be0cd3fd139cf650\\setup.py'"'"'; __file__='"'"'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pip-install-p2vpppra\\torch-cluster_2f5e4beb6db144f8be0cd3fd139cf650\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\runneradmin\AppData\Local\Temp\pip-record-arrcf94j\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Miniconda\envs\test\Include\torch-cluster'
         cwd: C:\Users\runneradmin\AppData\Local\Temp\pip-install-p2vpppra\torch-cluster_2f5e4beb6db144f8be0cd3fd139cf650\
    Complete output (76 lines):
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.8
    creating build\lib.win-amd64-3.8\torch_cluster
    copying torch_cluster\fps.py -> build\lib.win-amd64-3.8\torch_cluster
    copying torch_cluster\graclus.py -> build\lib.win-amd64-3.8\torch_cluster
    copying torch_cluster\grid.py -> build\lib.win-amd64-3.8\torch_cluster
    copying torch_cluster\knn.py -> build\lib.win-amd64-3.8\torch_cluster
    copying torch_cluster\nearest.py -> build\lib.win-amd64-3.8\torch_cluster
    copying torch_cluster\radius.py -> build\lib.win-amd64-3.8\torch_cluster
    copying torch_cluster\rw.py -> build\lib.win-amd64-3.8\torch_cluster
    copying torch_cluster\sampler.py -> build\lib.win-amd64-3.8\torch_cluster
    copying torch_cluster\__init__.py -> build\lib.win-amd64-3.8\torch_cluster
    running egg_info
    writing torch_cluster.egg-info\PKG-INFO
    writing dependency_links to torch_cluster.egg-info\dependency_links.txt
    writing requirements to torch_cluster.egg-info\requires.txt
    writing top-level names to torch_cluster.egg-info\top_level.txt
    reading manifest file 'torch_cluster.egg-info\SOURCES.txt'
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\runneradmin\AppData\Local\Temp\pip-install-p2vpppra\torch-cluster_2f5e4beb6db144f8be0cd3fd139cf650\setup.py", line 98, in <module>
        setup(
      File "C:\Miniconda\envs\test\lib\site-packages\setuptools\__init__.py", line 153, in setup
        return distutils.core.setup(**attrs)
      File "C:\Miniconda\envs\test\lib\distutils\core.py", line 148, in setup
        dist.run_commands()
      File "C:\Miniconda\envs\test\lib\distutils\dist.py", line 966, in run_commands
        self.run_command(cmd)
      File "C:\Miniconda\envs\test\lib\distutils\dist.py", line 985, in run_command
        cmd_obj.run()
      File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\install.py", line 61, in run
        return orig.install.run(self)
      File "C:\Miniconda\envs\test\lib\distutils\command\install.py", line 545, in run
        self.run_command('build')
      File "C:\Miniconda\envs\test\lib\distutils\cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "C:\Miniconda\envs\test\lib\distutils\dist.py", line 985, in run_command
        cmd_obj.run()
      File "C:\Miniconda\envs\test\lib\distutils\command\build.py", line 135, in run
        self.run_command(cmd_name)
      File "C:\Miniconda\envs\test\lib\distutils\cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "C:\Miniconda\envs\test\lib\distutils\dist.py", line 985, in run_command
        cmd_obj.run()
      File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\build_py.py", line 46, in run
        self.build_package_data()
      File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\build_py.py", line 103, in build_package_data
        for package, src_dir, build_dir, filenames in self.data_files:
      File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\build_py.py", line 55, in __getattr__
        self.data_files = self._get_data_files()
      File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\build_py.py", line 67, in _get_data_files
        self.analyze_manifest()
      File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\build_py.py", line 121, in analyze_manifest
        self.run_command('egg_info')
      File "C:\Miniconda\envs\test\lib\distutils\cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "C:\Miniconda\envs\test\lib\distutils\dist.py", line 985, in run_command
        cmd_obj.run()
      File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\egg_info.py", line 299, in run
        self.find_sources()
      File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\egg_info.py", line 306, in find_sources
        mm.run()
      File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\egg_info.py", line 541, in run
        self.add_defaults()
      File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\egg_info.py", line 585, in add_defaults
        self.read_manifest()
      File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\sdist.py", line 188, in read_manifest
        self.filelist.append(line)
      File "C:\Miniconda\envs\test\lib\site-packages\setuptools\command\egg_info.py", line 483, in append
        path = convert_path(item)
      File "C:\Miniconda\envs\test\lib\distutils\util.py", line 122, in convert_path
        raise ValueError("path '%s' cannot be absolute" % pathname)
    ValueError: path '/Users/rusty1s/github/pytorch_cluster/csrc/fps.cpp' cannot be absolute
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Miniconda\envs\test\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pip-install-p2vpppra\\torch-cluster_2f5e4beb6db144f8be0cd3fd139cf650\\setup.py'"'"'; __file__='"'"'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pip-install-p2vpppra\\torch-cluster_2f5e4beb6db144f8be0cd3fd139cf650\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\runneradmin\AppData\Local\Temp\pip-record-arrcf94j\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Miniconda\envs\test\Include\torch-cluster' Check the logs for full command output.
Error: Process completed with exit code 1.```

## Additional context
@bobturneruk No issue triggered?

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -39,9 +39,9 @@ jobs:
           pip install torch==1.7.1+cpu -f https://download.pytorch.org/whl/torch_stable.html
           pip install torchvision==0.8.2+cpu -f https://download.pytorch.org/whl/torch_stable.html
           pip install torch-scatter==2.0.7 -f https://pytorch-geometric.com/whl/torch-1.7.0+cpu.html
-          pip install torch-sparse -f https://pytorch-geometric.com/whl/torch-1.7.0+cpu.html
+          pip install torch-sparse==0.6.12 -f https://pytorch-geometric.com/whl/torch-1.7.0+cpu.html
           pip install torch-cluster==1.5.9 -f https://pytorch-geometric.com/whl/torch-1.7.0+cpu.html
-          pip install torch-spline-conv -f https://pytorch-geometric.com/whl/torch-1.7.0+cpu.html
+          pip install torch-spline-conv==1.2.1 -f https://pytorch-geometric.com/whl/torch-1.7.0+cpu.html
           pip install torch-geometric==1.7.0 --ignore-installed certifi
           pip install -e .[dev]
         shell: bash -l {0}

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -40,7 +40,7 @@ jobs:
           pip install torchvision==0.8.2+cpu -f https://download.pytorch.org/whl/torch_stable.html
           pip install torch-scatter==2.0.7 -f https://pytorch-geometric.com/whl/torch-1.7.0+cpu.html
           pip install torch-sparse -f https://pytorch-geometric.com/whl/torch-1.7.0+cpu.html
-          pip install torch-cluster -f https://pytorch-geometric.com/whl/torch-1.7.0+cpu.html
+          pip install torch-cluster==1.5.9 -f https://pytorch-geometric.com/whl/torch-1.7.0+cpu.html
           pip install torch-spline-conv -f https://pytorch-geometric.com/whl/torch-1.7.0+cpu.html
           pip install torch-geometric==1.7.0 --ignore-installed certifi
           pip install -e .[dev]

